### DISCUSSION TITLE
Access rules for setters and getters

### DISCUSSION THREAD

**Discussion by eon-s** [godotengine/godot-proposals#4520] [Scripting]
Haxe has an interesting approach to access options for setters and getters, even when there's public and protected keywords for visibility, access control can add more options.

```haxe
public var x(default, set):Int;

function set_x(newX) {
  return x = newX;
}
```

Details:
https://haxe.org/manual/class-field-property.html

Using that kind of rules may be enough to cover the needs of Godot games, frameworks and plugins.

Related proposal:
#2979

**Comment by azur-wolve** [comment]
access specifiers is something that I miss in GDScript, for good code  OOP design
